tests/integration/test_integration.py

```diff
@@ -90,3 +165,32 @@ async def test_given_application_is_deployed_when_scale_down_then_status_is_acti
             timeout=1000,
             wait_for_exact_units=num_units,
         )
+
+    @pytest.mark.abort_on_fail
+    async def test_given_traefik_deployed_when_relate_to_ingress_then_status_is_active(
+        self, ops_test: OpsTest, build_and_deploy, deploy_traefik
+    ):
+        await ops_test.model.add_relation(  # type: ignore[union-attr]
+            relation1=f"{APPLICATION_NAME}:ingress", relation2=f"{TRAEFIK_APP_NAME}:ingress"
+        )
+        await ops_test.model.wait_for_idle(  # type: ignore[union-attr]
+            apps=[APPLICATION_NAME, TRAEFIK_APP_NAME],
+            status="active",
+            timeout=1000,
+        )
+
+    @pytest.mark.abort_on_fail
+    async def test_given_related_to_traefik_when_vault_status_checked_then_returns_400(
```

Comment:
why not "401 Unauthorized"?

Reply:
After fixing the routing problem in Traefik, now it returns 500 because of TLS handshake error.